shims/spark311cdh/src/main/scala/com/nvidia/spark/rapids/shims/spark311cdh/Spark311CDHShims.scala

```diff
@@ -41,7 +41,7 @@ class Spark311CDHShims extends SparkBaseShims {
       GpuOverrides.exec[InMemoryTableScanExec](
         "Implementation of InMemoryTableScanExec to use GPU accelerated Caching",
         ExecChecks((TypeSig.commonCudfTypes + TypeSig.DECIMAL_64 + TypeSig.STRUCT
-            + TypeSig.ARRAY).nested().withPsNote(TypeEnum.DECIMAL,
+            + TypeSig.ARRAY + TypeSig.MAP).nested().withPsNote(TypeEnum.DECIMAL,
```

Comment:
Curious why there's an override for `InMemoryTableScanExec` in the CDH shim yet it's also overridden in the v2-style shims in sql-plugin.  CDH build should be using 311until320-nondb and therefore should already be picking up a mapping for this override.

Reply:
I think that makes sense, we shouldn't have the override for `InMemoryTableScanExec`. @gerashegalov are we reading this correctly? Spark311CDHShims extends SparkBaseShims which already has the `InMemoryTableScanExec` override.

Reply:
override only if necessary for CDH. there is significant code  duplication for PCBS.  Feel also free to introduce a finer-grained rule overrides so you don't have to deal with all execs at the same time https://github.com/NVIDIA/spark-rapids/commit/1441f308821ae7abf0d94589bd3c51c28e9cdac3#diff-8926565c5f6bc26203bed459ac37ea0fba15f9f5488aa2412aeac26b31f80ea6R3601

Reply:
Does the CI test the CDH shim?

Reply:
There currently isn't nightly testing of the CDH shim on our part (I believe Cloudera does this internally).  The CDH shim is built as part of the premerge.

The override in the CDH shim is almost the same source code as in the 311until320-nondb codepath except for the following diffs:
- CDH shim has a type checks and psnote about negative decimal scales which is missing in the other (oversight?)
- CDH shim has code to check serializer class if it's not the right instance and throws an illegal state exception

All of those diffs seem like they should be made consistent one way or the other and don't appear to be specific to CDH.  Therefore I don't see why the override in the CDH shim is needed.

Reply:
Yes, the differences are an oversight. Let me see what I can about the code duplication.